### PR TITLE
Add README to cron job with chapter for ilCronJobResult

### PR DESCRIPTION
**Pull request by legionth**
Adds a README with basic description of `ilCronJobResult`

### CODE REVIEW

**Comment by legionth** [comment]
Ping @mjansenDatabay

**Comment by Amstutz** [comment]
Thanks a lot!

File: Services/Cron/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Cron
+
+**Table of Contents**
+* [ilCronJobResult](#ilCronJobResult)
+
+## ilCronJobResult
+
+The class `ilCronJobResult` determine the status
```
Comment (mjansenDatabay) [inline Services/Cron/README.md]: ... determines

File: Services/Cron/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+of a current cron job.
+
+The status are:
+* `STATUS_INVALID_CONFIGURATION`
+  This status will indicate the current configuration
+  for this cron job is not correct and MUST be
+  adjusted.
+* `STATUS_NO_ACTION`
+  This status will indicate that the cron job
```
Comment (mjansenDatabay) [inline Services/Cron/README.md]: This status will indicate that the cron job did not perform any action.

Examples
* A cron job sending emails decided to send no emails at all.
* A cron job responsible for deleting orphaned objects did not find any object to delete.
* The lucene cron job decided that the index does not require an update.
* etc.

File: Services/Cron/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  was not executed.
+* `STATUS_OK`
+  This status indicates that the cron job
+  has been successfully finished.
+* `STATUS_CRASHED`
+  A critical failure has occurred during
+  the execution of the cron job, which led
+  to an critical error.
+  The cron job needs to be restarted by
```
Comment (mjansenDatabay) [inline Services/Cron/README.md:34]: Please verifiy the described behaviour.

File: Services/Cron/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  } catch (Exception $exception) {
+    $result->setStatus(ilCronJobResult::STATUS_FAIL);
+    $result->setMessage($exception->getMessage());
+  }
+
+  return $result;
+}
+```
+
+A message can be added additionally that will be
```
Comment (mjansenDatabay) [inline Services/Cron/README.md]: A message SHOULD be added ..

File: Services/Cron/README.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# Cron
+
```
Comment (mjansenDatabay) [inline Services/Cron/README.md:2]: Maybe we should add some RFC introduction here.

The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.

**Review by mjansenDatabay — CHANGES_REQUESTED** [review]

**Comment by legionth** [comment]
Added your recommendations in the latest commit. Still need to verfiy the critical crash behavior. I will set this to WIP as long as I didn't verified this.

File: Services/Cron/README.md (hunk 3 above)
Comment (legionth) [reply Services/Cron/README.md:34]: I've been able to verify the described behavior. I will remove the `WIP` tag.

**Comment by legionth** [comment]
Ping @mjansenDatabay . Have been able to verify that the cron job must be restarted on the status `STATUS_CRASHED`.

**Review by mjansenDatabay — APPROVED** [review]

**Pull request merged**